src/relentless/optimize/method.py

```diff
@@ -351,6 +360,15 @@ def optimize(self, objective, design_variables, directory=None):
 
         if directory is not None:
             directory = data.Directory.cast(directory, create=mpi.world.rank_is_root)
+            if len(os.listdir(directory.path)) != 0:
+                if overwrite is True:
+                    directory.clear_contents()
```

Comment:
I think this needs to be guarded so that only the root rank calls it. It will be safe to not put another barrier before it because we doing `create` on the root only.
```suggestion
                    if mpi.world.rank_is_root:
                        directory.clear_contents()
```